presto-orc/src/main/java/com/facebook/presto/orc/OrcReader.java

```diff
@@ -186,6 +215,59 @@ public OrcReader(
         }
 
         this.cacheable = requireNonNull(cacheable, "hiveFileContext is null");
+
+        Optional<DwrfStripeCache> dwrfStripeCache = Optional.empty();
+        if (orcFileTail.getDwrfStripeCacheData().isPresent() && footer.getDwrfStripeCacheOffsets().isPresent()) {
+            DwrfStripeCache cache = buildDwrfStripeCache(footer.getStripes(), orcFileTail.getDwrfStripeCacheData().get(), footer.getDwrfStripeCacheOffsets().get());
+            dwrfStripeCache = Optional.of(cache);
+        }
+
+        requireNonNull(stripeMetadataSourceFactory, "stripeMetadataSourceFactory is null");
+        this.stripeMetadataSource = requireNonNull(stripeMetadataSourceFactory.create(dwrfStripeCache), "stripeMetadataSource is null");
+    }
+
+    private DwrfStripeCache buildDwrfStripeCache(List<StripeInformation> stripes, DwrfStripeCacheData dwrfStripeCacheData, List<Integer> cacheOffsets)
+    {
+        List<StripeInformation> sortedStripes = new ArrayList<>(stripes);
+        sortedStripes.sort(comparingLong(StripeInformation::getOffset));
```

Comment:
This is a bit controversial, but it's coming from https://github.com/prestodb/presto/blob/master/presto-orc/src/main/java/com/facebook/presto/orc/AbstractOrcRecordReader.java#L194-L203. I don't know what is the real reason for sorting, it's a bit hard to do blame on this file because it was refactored a couple times.

We don't need sorting in FB use cases, I'm open to nuke sorting from this place.

Reply:
Can we please add a comment to capture this reason ?

Reply:
Done, it's not very convincing thou :)